### PR TITLE
Add universal preloader function

### PR DESCRIPTION
**Pull request by pkfranz**
@mikelmaron This should seamlessly work on any project, regardless of `chapter.id` naming conventions. 

Possible feature add: Adding it as an boolean variable in the array vs. always on?

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -334,7 +376,18 @@
     })
     .onStepEnter(async response => {
         var chapter = config.chapters.find(chap => chap.id === response.element.id);
+        var chapterid = chapter.id;
```
Comment (mikelmaron) [inline src/index.html:379]: where is chapterid used?

File: src/index.html [hunk 2]
```diff
@@ -334,7 +376,18 @@
...
         response.element.classList.add('active');
+
+        // Code for preloader, app will not show an active map until underlying tilesets are fully loaded
+
+        if (chapter.id != null) {
```
Comment (mikelmaron) [inline src/index.html:384]: whether or not chapter.id is null, opacity is set to 0

File: src/index.html [hunk 3]
```diff
@@ -457,6 +510,18 @@
     insetMap.getSource('boundsSource').setData(bounds);
 }
 
+function customizeChapters(chap) {
```
Comment (mikelmaron) [inline src/index.html:513]: this function is never called?

**Comment by mikelmaron** [comment]
@pkfranz some questions in comments. also why is config.js.template renamed?

**Comment by chriswhong** [comment]
Closing as stale

**Pull request closed without merging**